File: lua5.4/iniciante/1837.lua
function euclideanDivision(a, b)
    local limit = b < 0 and -b or b
    local r, q = 0, 0

    for r = 0, limit - 1 do
        if (a - r) % b == 0 then
            q = (a - r) // b
            return q, r
        end
    end

    return q, r
end

local a, b = io.read("*n", "*n")
local q, r = euclideanDivision(a, b)
print(q .. " " .. r)

File: php/iniciante/1837.php
<?php
function euclideanDivision($a, $b) {
    $limit = $b < 0 ? -$b : $b;
    $r = 0;
    $q = 0;

    for ($r = 0; $r < $limit; $r++) {
        if (($a - $r) % $b == 0) {
            $q = ($a - $r) / $b;
            break;
        }
    }

    return array($q, $r);
}

list($a, $b) = explode(" ", fgets(STDIN));
$a = intval($a);
$b = intval($b);

list($q, $r) = euclideanDivision($a, $b);
echo "$q $r\n";
?>
